pkg/process/checks/chunking.go

```diff
@@ -0,0 +1,168 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package checks
+
+import (
+	"unsafe"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+)
+
+type payloadList interface {
+	Len() int
+	WeightAt(idx int) int
+	ToChunk(start, end int, weight int)
+}
+
+type chunkAllocator interface {
+	TakenSize() int
+	TakenWeight() int
+	Append()
+	Next()
+}
+
+func chunkPayloadsBySizeAndWeight(l payloadList, a chunkAllocator, maxChunkSize int, maxChunkWeight int) {
+	start := 0
+	chunkWeight := 0
+	availableSize := maxChunkSize - a.TakenSize()
+	availableWeight := maxChunkWeight - a.TakenWeight()
+	for i := 0; i < l.Len(); i++ {
+		itemWeight := l.WeightAt(i)
+		size := i - start
+		skipItem := false
+		if size >= availableSize || chunkWeight+itemWeight > availableWeight {
+			if size > 0 {
+				l.ToChunk(start, i, chunkWeight)
+				a.Next()
+			}
+
+			chunkWeight = 0
+			start = i
+			if itemWeight >= maxChunkWeight {
+				if availableWeight < maxChunkWeight {
+					a.Append()
+				}
+				l.ToChunk(i, i+1, itemWeight)
+				a.Next()
+				start = i + 1
+				skipItem = true
+			} else {
+				for maxChunkSize-a.TakenSize() < 1 || maxChunkWeight-a.TakenWeight() < itemWeight {
+					a.Next()
+				}
+			}
+			availableSize = maxChunkSize - a.TakenSize()
+			availableWeight = maxChunkWeight - a.TakenWeight()
+		}
+		if !skipItem {
+			chunkWeight += itemWeight
+		}
+	}
+	if start < l.Len() {
+		l.ToChunk(start, l.Len(), chunkWeight)
+	}
+}
+
+func chunkProcessesBySizeAndWeight(procs []*model.Process, maxChunkSize, maxChunkWeight int, chunker *collectorProcChunker) {
+	list := &processList{
+		procs:   procs,
+		chunker: chunker,
+	}
+	chunkPayloadsBySizeAndWeight(list, chunker, maxChunkSize, maxChunkWeight)
+}
+
+type processList struct {
+	procs   []*model.Process
+	chunker processChunker
+}
+
+type processChunker interface {
+	Accept(procs []*model.Process, weight int)
+}
+
+func (l *processList) Len() int {
+	return len(l.procs)
+}
+
+func (l *processList) WeightAt(idx int) int {
+	if idx >= len(l.procs) {
+		return 0
+	}
+	return weighProcess(l.procs[idx])
+}
+
+func (l *processList) ToChunk(start, end int, weight int) {
+	l.chunker.Accept(l.procs[start:end], weight)
+}
+
+type chunkProps struct {
+	weight int
+	size   int
+}
+type chunkPropsTracker struct {
+	props []chunkProps
+	idx   int
+}
```

Comment:
what do you think about calling these ?
```golang
type chunk struct {
	weight int
	size   int
}
type chunkTracker struct {
	chunks []chunk
	idx   int
}
```
`weight` and `size` are properties from a `chunk` so I'm not sure if we need to call the struct `chunkProps`. What do you think?